### DISCUSSION TITLE
About BDS-3 code biases.

### DISCUSSION THREAD

**Discussion by gnssxx** [groops-devs/groops#140] [Q&A] [answered]
Hello, According to your paper, we can follow the IGS convention of  defining the clock datum based on the GPS C1W and C2W signals when estimating the GPS code biases. Does the following code complete this work.
![tmp23A3](https://user-images.githubusercontent.com/75126376/206981254-eff8ca0c-b1e4-4a52-986a-f5b14e5bd623.png)
Do you set the datum when estimating the BDS-3 code biases?  I can't find the relevant code here.

**Comment by zhedumi** [comment, marked as answer]
Hi,

You are on the correct code lines. As described in chapter 6.5.3.1 p. 107 within [Strasser 2022](https://pure.tugraz.at/ws/portalfiles/portal/40917089/Strasser_2022_Doctoral_Thesis_Published_Version.pdf) by setting the C1W and C2W to 0 the nullspace is defined over those two observation type.

In our approach we simulate the observation equations, creating the normal equations and proceed to create estimable linear combinations of the code biases by eigenvalue decomposition of the simulated normal equations. The eigenvectors are classified into a nullspace matrix and our transformation matrix responsible for generating the estimable linear combinations. For GPS we selectively set C1W and C2W to be our nullspace since this is IGS convention. This is done in the code that you put in here. 
For Galileo on the otherhand and more extended for BeiDou there is no such convention for the biases. 
And in Galileos case the stations around the globe either measure C?Q/C, or C?X which results in  "split network" leading to a group of receiver available for bias estimation for C?X and a group of receivers available for bias estimation of C?Q/C. This in combination of our pseudo absolute base estimation by null constraining can lead to abritary differences between I/Q. 

For BDS we also do the same approach as with Galileo.

Best regards,
Patrick
